pkg/collector/corechecks/cluster/orchestrator/processors/common/base.go

```diff
@@ -53,3 +55,24 @@ func ExtractModelManifests(ctx processors.ProcessorContext, resourceManifests []
 	}
 	return cm
 }
+
+// K8sAgentInfo returns the agent info for k8s
+func K8sAgentInfo(i *installinfo.InstallInfo) *model.K8SAgentInfo {
```

Comment:
Agent info should be already sent part of the Agent host metadata information. 
From what I know, this info is store in REDAPL and already use by fleet automation. 
Have you considered using this info already available in REDAPL?

Reply:
The agent info data concerning the cluster-agent in not available so we would still need to collect it in the agent for the cluster resource